schemas/v1/schema.go

```diff
@@ -1366,3 +1371,63 @@ CREATE INDEX state_heights_height_index ON {{ .SchemaName | default "public"}}.s
 CREATE INDEX state_heights_parentstateroot_index ON {{ .SchemaName | default "public"}}.state_heights USING btree (parentstateroot);
 
 `
+
+// TODO(frrist): I shoe-horned this in here in a bout of confusion and frustration with circular deps.
+// We need to define a location for the schema config so we can apply it to all migrations. Or I'm just holding it wrong..
+// batch is a syntactic helper for registering a migration
+func batch(sqls ...string) func(db migrations.DB) error {
+	return func(db migrations.DB) error {
+		for _, sql := range sqls {
+			tmpl, err := template.New("base").Funcs(schemaTemplateFuncMap).Parse(sql)
+			if err != nil {
+				return xerrors.Errorf("parse base template: %w", err)
+			}
+			var buf strings.Builder
+			cfg := schemas.Config{
+				SchemaName: "public",
+			}
+			if err := tmpl.Execute(&buf, cfg); err != nil {
+				return xerrors.Errorf("execute base template: %w", err)
+			}
+			if _, err := db.Exec(buf.String()); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+var schemaTemplateFuncMap = template.FuncMap{
+	"default": func(def interface{}, value interface{}) interface{} {
+		if isEmpty(value) {
+			return def
+		}
+		return value
+	},
+}
+
+func isEmpty(val interface{}) bool {
+	v := reflect.ValueOf(val)
+	if !v.IsValid() {
+		return true
+	}
+
+	switch v.Kind() {
+	case reflect.Array, reflect.Slice, reflect.Map, reflect.String:
+		return v.Len() == 0
+	case reflect.Bool:
+		return !v.Bool()
+	case reflect.Complex64, reflect.Complex128:
+		return v.Complex() == 0
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return v.Int() == 0
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr:
+		return v.Uint() == 0
+	case reflect.Float32, reflect.Float64:
+		return v.Float() == 0
+	case reflect.Struct:
+		return false
+	default:
+		return v.IsNil()
+	}
+}
```

Comment:
I took the easy way out and copy-pasted some lines from the storage model to get this to create valid migrations for the new models. @iand how should we be specifying Schema names in the migration?

Reply:
See https://github.com/filecoin-project/sentinel-visor/pull/578 for migration capability